QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java

```diff
@@ -47,11 +58,21 @@ public void event() {
 					angle += π;
 				}
 			}
+			
+			if (fieldCentric.get()) {
+				double centric = calibratedAngle.get() - centricAngleOffset.get();
+				double currentAngle = HeadingSensor.yaw.get();
+				currentAngle = currentAngle / 180 * π;
+				currentAngle -= centric;
+				angle -= currentAngle;
+			}
+			
 			float leftFront = (float) (speed * Math.sin(angle - π / 4) - rotationspeed);
 			float rightFront = (float) (speed * Math.cos(angle - π / 4) + rotationspeed);
 			float leftBack = (float) (speed * Math.cos(angle - π / 4) - rotationspeed);
 			float rightBack = (float) (speed * Math.sin(angle - π / 4) + rotationspeed);
-			float normalize = Math.max(Math.max(Math.abs(leftFront), Math.abs(rightFront)), Math.max(Math.abs(leftBack), Math.abs(rightBack)));
+			float normalize = Math.max(Math.max(Math.abs(leftFront), Math.abs(rightFront)),
+					Math.max(Math.abs(leftBack), Math.abs(rightBack)));
```

Comment:
You may want to indent the second line here so that it's clearer what's happening.